src/esp/assets/ResourceManager.cpp

```diff
@@ -1110,6 +1111,106 @@ bool ResourceManager::loadGeneralMeshData(
   return true;
 }  // loadGeneralMeshData
 
+bool ResourceManager::loadTrajectoryVisualization(
+    const std::string& trajVisName,
+    const std::vector<Mn::Vector3>& pts,
+    int numSegments,
+    float radius,
+    bool smooth,
+    int numInterp) {
+  // enforce required minimum/reasonable values if illegal values specified
+  if (numSegments < 3) {  // required by circle prim
+    numSegments = 3;
+  }
+  if (smooth && (numInterp <= 0)) {  // 10 points per trajectory point
+    numInterp = 10;
+  }
+  if (radius <= 0) {  // 1 mm radius minimum
+    radius = .001;
+  }
+
+  LOG(INFO) << "ResourceManager::loadTrajectoryVisualization : Calling "
+               "trajectoryTubeSolid to build a tube named :"
+            << trajVisName << " with " << pts.size()
+            << " points, building a tube of radius :" << radius << " using "
+            << numSegments << " circular segments and " << numInterp
+            << " interpolated points between each trajectory point.";
+
+  // get circular vertices to use to construct the trajectory tube
+  // make sure importer is open before use
+  primitiveImporter_->openData("");
+  // configuration for PrimitiveImporter - replace appropriate group's data
+  // before instancing prim object
+  Cr::Utility::ConfigurationGroup* cfgGroup =
+      primitiveImporter_->configuration().group("circle3DWireframe");
+  CORRADE_ASSERT(cfgGroup != nullptr,
+                 "ResourceManager::loadTrajectoryVisualization: Error with "
+                 "primitiveImporter_ : Failed to supply circle3DWireframe "
+                 "configuration. Aborting.",
+                 false);
+
+  // set value for config group for 3D circle wireframes
+  cfgGroup->setValue<int>("segments", numSegments);
+
+  // get verts for circle primitive to use as endpoints and transform them to
+  // more easily use them to calculate tangents at appropriate raidius and
+  // normals
+  Cr::Containers::Array<Magnum::Vector3> circleVerts =
+      primitiveImporter_->mesh("circle3DWireframe")->positions3DAsArray();
```

Comment:
Let's peel the onion a bit and sidestep unnecessary interfaces -- this all could be just a single expression :stuck_out_tongue_closed_eyes: 

```suggestion
  Cr::Containers::Array<Magnum::Vector3> circleVerts =
      Mn::Primitives::circle3DWireframe(numSegments).positions3DAsArray();
```

Reply:
Glorious! Ty :)